c2cgeoportal/views/pdfreport.py

```diff
@@ -0,0 +1,162 @@
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+# The views and conclusions contained in the software and documentation are those
+# of the authors and should not be interpreted as representing official policies,
+# either expressed or implied, of the FreeBSD Project.
+
+from pyramid.view import view_config
+from pyramid.response import Response
+from pyramid.httpexceptions import HTTPBadRequest, HTTPBadGateway, HTTPForbidden
+
+import httplib2
+from urlparse import urlparse
+from json import dumps
+
+from c2cgeoportal.lib.filter_capabilities import get_protected_layers, \
+    get_private_layers
+
+import logging
+log = logging.getLogger(__name__)
+
+
+class PdfReport:
+
+    def __init__(self, request):
+        self.request = request
+        self.config = self.request.registry.settings.get('pdfreport', {})
+
+    def do_print(self, spec):
+        """ Get created PDF. """
+        url = self.config['print_url'] + "/print/buildreport.pdf"
+        http = httplib2.Http()
+        h = dict(self.request.headers)
+        h["Content-Type"] = "application/json"
+        if urlparse(url).hostname != 'localhost':
+            h.pop('Host')
+        try:
+            resp, content = http.request(
+                url, method='POST', headers=h, body=dumps(spec)
+            )
+        except:
+            return HTTPBadGateway()
+
+        headers = {}
+        if 'content-type' in resp:
+            headers['content-type'] = resp['content-type']
+        if 'content-disposition' in resp:
+            headers['content-disposition'] = resp['content-disposition']
+        return Response(
+            content, status=resp.status, headers=headers
+        )
+
+    @view_config(route_name='pdfreport', renderer="json")
+    def getreport(self):
+        id = int(self.request.matchdict['id'])
+        layername = self.request.matchdict['layername']
+        if id is None or layername is None:
+            raise HTTPBadRequest
+        featureid = layername + "." + str(id)
+
+        # check user credentials
+        role_id = None if self.request.user is None else \
+            self.request.user.role.id
+
+        # FIXME: support of mapserver groups
+        if layername in get_private_layers() and \
+                layername not in get_protected_layers(role_id):
+            raise HTTPForbidden
+
+        params = {
+            'service': 'WFS',
+            'version': '1.0.0',
+            'request': 'GetFeature',
+            'typeName': layername,
+            'featureid': featureid,
+            'outputFormat': 'geojson'
+        }
+        srs = self.config.get('srs_name')
+        if srs is not None:
+            params['srsName'] = srs
+
+        request_url = self.config['mapserv_url'] + "?" \
+            + "&".join(["%s=%s" % i for i in params.items()])
+
+        baselayer = self.config.get("layers", {}). \
+            get(layername, {}).get('baselayers')
```

Comment:
`mapserv_url` should be the same as `request.get_route('mapserv_proxy')`
`print_url` already exists can we reuse it ?

Otherwise that looks good :-)